packages/sol-meta/src/constructor.ts

```diff
@@ -0,0 +1,80 @@
+import * as S from 'solidity-parser-antlr';
+
+import * as utils from './utils';
+
+export interface ConstructorArguments {
+    [contractName: string]: utils.Litteral[];
+}
+
+export function makeConstructor(consArgs: ConstructorArguments): S.FunctionDefinition {
+    // TODO: Only include actually used constructors.
+    return {
+        type: S.NodeType.FunctionDefinition,
+        name: null,
+        parameters: {
+            type: S.NodeType.ParameterList,
+            parameters: [],
+        },
+        returnParameters: null,
+        visibility: S.Visibility.Public,
+        stateMutability: S.StateMutability.Default,
+        modifiers: Object.keys(consArgs).map<S.ModifierInvocation>(name => ({
+            type: S.NodeType.ModifierInvocation,
+            name,
+            arguments: consArgs[name].map(utils.litteral),
+        })),
+        isConstructor: true,
+        body: {
+            type: S.NodeType.Block,
+            statements: [],
+        },
+    };
+}
+
+// Solidity by itself does not give an error when a contract is unintentionally
+// abstract. We can force Solidity to produce an error and point us to the
+// abstract function by trying to runtime instantiate the contract. This function
+// produces a small contract that tries to instantiate the given contract.
+export function nonAbstractForcer(contractName: string): S.ContractDefinition {
```

Comment:
It might be too late for that comment, but wouldn't it be more readable if we use a completely different approach?
Instead of constructing the AST we construct solidity using the template strings and then parse it to AST.
I just saw that the commment text might be as well the code itself. ASTs are hard to read by humans and we already have a parser to convert.

Reply:
That's a reasonable approach. While writing I realized that it was probably better.

I did not go this route because it requires patching `soldity-parser-antlr` to parse fragments instead of whole contracts.

Another slight advantage is that the type system prevents you from accidentally producing incorrect code. Like the template `${a} * 3` will fail when `a = 4 + 4`.

Reply:
Good point.